state/caassequence.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2012, 2013 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"fmt"
+
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+// sequence safely increments a database backed sequence, returning
+// the next value.
+func (st *CAASState) sequence(name string) (int, error) {
+	sequences, closer := st.database.GetCollection(sequenceC)
```

Comment:
The current mjs/caas-prototype already has this done for sequenceWithMin, doing it for sequence is similar.